### PR TITLE
 fix: lint issues, add types back

### PR DESCRIPTION
**Pull request by JounQin**
related to https://github.com/developit/microbundle/issues/829

### CODE REVIEW

File: packages/mdx/types/index.d.ts [hunk 1]
```diff
@@ -11,35 +11,35 @@ declare namespace mdx {
     filepath?: string
 
     /**
-     * skip the addition of 'export default' statement when serializing
+     * Skip the addition of 'export default' statement when serializing
```
Comment (ChristianMurphy) [inline packages/mdx/types/index.d.ts:14]: Is this linting related?
I'm not seeing a related lint rule in the configuration

File: packages/loader/types/index.d.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
 // TypeScript Version: 3.4
 
-import {ComponentType} from 'react'
-
+// tslint:disable-next-line: no-single-declare-module
 declare module '*.mdx' {
+  import {ComponentType} from 'react'
```
Comment (ChristianMurphy) [inline packages/loader/types/index.d.ts:5]: out of curiosity, what is the benefit of moving the import inside the `declare module`?

File: package.json [hunk 3]
```diff
@@ -39,15 +39,16 @@
     "@babel/preset-react": "^7.12.0",
     "babel-loader": "^8.2.0",
     "babel-preset-gatsby-package": "^0.12.0",
-    "dtslint": "^4.0.0",
-    "eslint": "^7.13.0",
-    "eslint-config-prettier": "^7.0.0",
-    "eslint-config-xo": "^0.33.0",
+    "dtslint": "^4.0.8",
+    "eslint": "^7.23.0",
```
Comment (ChristianMurphy) [inline package.json:43]: these updates appear to already be/have been in semver range?

**Review by ChristianMurphy — APPROVED** [review]
Thanks @JounQin!

File: package.json (hunk 3 above)
Comment (JounQin) [reply package.json:43]: When debug an issue related to duplicate versions of `@typescript-eslint/eslint-plugin` due to `gatsby`, I tried to upgrade them to latest versions manually.

File: packages/mdx/types/index.d.ts (hunk 1 above)
Comment (JounQin) [reply packages/mdx/types/index.d.ts:14]: https://eslint.org/docs/rules/capitalized-comments

Maybe xo enables it, I'm not sure.

File: packages/loader/types/index.d.ts (hunk 2 above)
Comment (JounQin) [reply packages/loader/types/index.d.ts:5]: It is better to write declarations separately for each `module` IMO. Besides, the `disable` is not related.
